janus_server/src/aggregator/aggregation_job_creator.rs

```diff
@@ -279,6 +281,101 @@ impl<C: Clock + 'static> AggregationJobCreator<C> {
             })
             .await?)
     }
+
+    /// Look for combinations of client reports and collect job aggregation parameters that do not
+    /// yet have a report aggregation, and batch them into new aggregation jobs. This should only
+    /// be used with VDAFs that have non-unit type aggregation parameters.
+    // This is only used in tests thus far.
+    #[cfg(test)]
+    #[tracing::instrument(skip(self), err)]
+    async fn create_aggregation_jobs_for_task_with_param<const L: usize, A: vdaf::Aggregator<L>>(
+        &self,
+        task: &Task,
+    ) -> anyhow::Result<()>
+    where
+        for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
+        A::PrepareMessage: Send + Sync,
+        A::PrepareState: Send + Sync + Encode,
+        A::OutputShare: Send + Sync,
+        for<'a> &'a A::OutputShare: Into<Vec<u8>>,
+        A::AggregationParam: Send + Sync + Eq + Hash,
```

Comment:
Dubious idea: `create_aggregation_jobs_for_task_no_param` has a bound `A: vdaf::Aggregator<L, AggregationParam = ()>` ensuring it can only be used where the aggregation parameter is `()`. It'd be cool if we could ensure that `create_aggregation_jobs_for_task_with_param` is only used where `A::AggregationParam != ()`. Now, that's not possible out of the box in Rust, but what if we defined some new marker trait like `VdafWithAggregationParameter` and then "implemented" that on `prio::vdaf::Poplar1` and our test VDAF types? Then I think the compiler could ensure we don't accidentally use this path.